pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -397,16 +412,27 @@ func (gc *GarbageCollector) attemptToDeleteWorker(ctx context.Context, item inte
 func (gc *GarbageCollector) isDangling(ctx context.Context, reference metav1.OwnerReference, item *node) (
 	dangling bool, owner *metav1.PartialObjectMetadata, err error) {
 
+	logger := klog.FromContext(ctx)
+	logger = klog.LoggerWithValues(logger,
+		"object.uid", item.identity.UID,
+		"ownerReference.apiVersion", reference.APIVersion,
+		"ownerReference.kind", reference.Kind,
+		"ownerReference.name", reference.Name,
+	)
+
```

Comment:
However, if ContextualLogging feature gate is not enabled, these keyValues pairs are lost and cannot be rolled back to default.

/cc @pohly

Reply:
Is that true? Without contextual logging enabled, `klog.FromContext()` will return a default sentinel value, but it's still a structural logger. This PR does not use `klog.NewContext()` to pass information through the logger in the context, and *that* would be the only bit that doesn't work without the feature flag, right?

Reply:
```
func LoggerWithValues(logger Logger, kv ...interface{}) Logger {
	if contextualLoggingEnabled {
		return logger.WithValues(kv...)
	}
	return logger
}
```

😢

Reply:
Yes, we had to make these functions a NOP to satisfy the feature gate criteria for alpha features. For these conversions that means that we can start to use some of the new features to make logs better, but it won't have an effect unless the feature gets enabled.

That also implies that the values still need to be logged below. They will only get logged once when the feature is enabled.

Reply:
@pohly this is pretty surprising to me - doesn't it mean that you can't use any contextual logging features whatsoever until the feature gates default to on... ?

Reply:
You can enable the feature gate via the usual `--feature-gates ContextualLogging=true`.

Reply:
Right, but I'm asking about the broader case - if we make a change to some part of the k8s code-base to add context to a logger and remove it from a structured log message, it will regress logging output for the default feature gate case, no?

Reply:
Or I guess @pohly what's the policy for "regression" on logging?

Reply:
> if we make a change to some part of the k8s code-base to add context to a logger and remove it from a structured log message

We can't do that in a single step. We first have to add contextual logging, then at a later point when it is considered "always on" (= beta or GA) the then redundant key/value pairs in the `logger.Info` or `logger.Error` can be removed.

The implementation intentionally filters out redundant key/value pairs when contextual logging is on and WithValues was used, to support such a transition period.

I know that it is annoying, but we had to support turning off the feature.

Reply:
> what's the policy for "regression" on logging?

Log messages can change at any time, there are no stability guarantees. But we obviously don't want to make logs less useful because of the migration to contextual logging, so values will have to be in both (`WithValues` and parameters) for a while.

Reply:
I just pushed an update that preserves

Reply:
Looks like I might have missed some - checking